### PR TITLE
Env effect FTW!

### PR DESCRIPTION
**Pull request by pmellati**

### CODE REVIEW

File: bored-robot.cabal [hunk 1]
```diff
@@ -32,6 +33,8 @@ library
                      , process-extras
                      , text
                      , vector
+                     , turtle
```
Comment (luke-clifton) [inline bored-robot.cabal]: Are you using turtle?

Comment (RobbieMcMichael) [inline bored-robot.cabal]: I think `turtle` was removed earlier and this PR is probably reintroducing it by accident.

File: bored-robot.cabal [hunk 2]
```diff
@@ -32,6 +33,8 @@ library
...
+                     , containers
```
Comment (RobbieMcMichael) [inline bored-robot.cabal]: Add `containers` in alphabetical order.

File: bored-robot.cabal [hunk 3]
```diff
@@ -73,6 +76,7 @@ test-suite bored-robot-test
                      , tasty-quickcheck
                      , text
                      , vector
+                     , containers
```
Comment (RobbieMcMichael) [inline bored-robot.cabal]: Add `containers` in alphabetical order.

File: bored-robot.cabal (hunk 1 above)
Comment (pmellati) [inline bored-robot.cabal]: Good find! Removed.

File: src/CI/Env.hs [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+          ( MemberU2 Lift (Lift IO) r )
+       => Eff (Env ': r) a
+       -> Eff r a
+runEnv = handleRelay return handle
+  where
+    handle :: Handler Env r a
+    handle (EnvRead key) k = lift (lookup key) >>= k
+
+    lookup :: Text -> IO (Maybe Text)
+    lookup key = fmap (fmap T.pack) $ lookupEnv $ T.unpack key
```
Comment (RobbieMcMichael) [inline src/CI/Env.hs]: We often prefer things like:
```Haskell
lookup = fmap (fmap T.pack) . lookupEnv . T.unpack
```

File: test/Spec.hs [hunk 5]
```diff
@@ -2,6 +2,7 @@ import qualified AppSpec
 import qualified CI.DockerSpec     as DockerSpec (test)
 import qualified CI.FilesystemSpec as FilesystemSpec (test)
 import qualified CI.ProcSpec       as ProcSpec (test)
+import qualified CI.EnvSpec        as EnvSpec  (test)
```
Comment (RobbieMcMichael) [inline test/Spec.hs]: Another one to reorder.

File: bored-robot.cabal (hunk 2 above)
Comment (pmellati) [inline bored-robot.cabal]: Done!

File: bored-robot.cabal (hunk 3 above)
Comment (pmellati) [inline bored-robot.cabal]: Done!

File: src/CI/Env.hs (hunk 4 above)
Comment (pmellati) [inline src/CI/Env.hs]: Done!

File: test/Spec.hs (hunk 5 above)
Comment (pmellati) [inline test/Spec.hs]: Done!

**Pull request merged**